fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -476,11 +476,11 @@ protected static void addCacheControlHeaders(final HttpServletResponse servletRe
 
         final FedoraResource mutableResource = resource instanceof NonRdfSourceDescription
                 ? ((NonRdfSourceDescription) resource).getDescribedResource() : resource;
-        final EntityTag etag = new EntityTag(mutableResource.getEtagValue());
+        final EntityTag etag = new EntityTag(mutableResource.getEtagValue(), true);
         final Date date = mutableResource.getLastModifiedDate();
 
         if (!etag.getValue().isEmpty()) {
-            servletResponse.addHeader("ETag", etag.toString());
+            servletResponse.addHeader("ETag", mutableResource.hasType(FEDORA_NON_RDF_SOURCE_DESCRIPTION) ? "" : "W/" ) + etag);
```

Comment:
The inclusion of `W/` should be handled by the `EntityTag` class, not inserted manually here.
Why exactly are you suggesting that non-rdf-source-descriptions not have weak etags?